lib/core/sessions.js

```diff
@@ -464,7 +488,7 @@ function endTransaction(session, commandName, callback) {
           e.errorLabels = [];
         }
 
-        if (isUnknownTransactionCommitResult(e)) {
+        if (isUnknownTransactionCommitResult(e) || isMaxTimeMSExpiredError(e)) {
```

Comment:
nit: what if we added a `isMaxTimeMSExpiredError` check inside of `isUnknownTransctionCommitResult`?